src/ViewLayer/RaceModeDisplay/RaceModeFaults/FaultList.cpp

```diff
@@ -0,0 +1,151 @@
+#include "FaultList.h"
+#include <QDebug>
+
+namespace
+{
+    int NUMBER_OF_MOTOR_ERRORS = 8;
+    int NUMBER_OF_MOTOR_LIMITS = 7;
+    //Motor errors
+    QString MOTOR_OVER_SPEED = "Motor Over Speed";
+    QString SOFTWARE_OVER_CURRENT = "Software Over Current";
+    QString DC_BUS_OVER_VOLTAGE = "DC Bus Over Voltage";
+    QString BAD_MOTOR_POSITION_HALL_SEQUENCE = "Bad Motor Position Hall Sequence";
+    QString WATCHDOG_CAUSED_LAST_RESET = "Watchdog Caused Last Reset";
+    QString CONFIG_READ_ERROR = "Config Read Error";
+    QString RAIL_UNDER_VOLTAGE_LOCK_OUT = "Rail Under-Voltage Lock Out";
+    QString DESATURATION_FAULT = "Desaturation Fault";
+    //Motor limits
+    QString OUTPUT_VOLTAGE_PWM_LIMIT = "Output Voltage PWM Limit";
+    QString MOTOR_CURRENT_LIMIT = "Motor Current Limit";
+    QString VELOCITY_LIMIT = "Velocity Limit";
+    QString BUS_CURRENT_LIMIT = "Bus Current Limit";
+    QString BUS_VOLTAGE_UPPER_LIMIT = "Bus Voltage Upper Limit";
+    QString BUS_VOLTAGE_LOWER_LIMIT = "Bus Voltage Lower Limit";
+    QString IPM_MOTOR_TEMPERATURE_LIMIT = "IPM or Motor Temperature Limit";
+
+    QString RED = "red";
+    QString YELLOW = "yellow";
+}
+
+FaultList::FaultList()
+    : errorList_(
+{
+    FaultLabel(MOTOR_OVER_SPEED, RED, 0, false)
+    , FaultLabel(SOFTWARE_OVER_CURRENT, RED, 0, false)
+    , FaultLabel(DC_BUS_OVER_VOLTAGE, RED, 0, false)
+    , FaultLabel(BAD_MOTOR_POSITION_HALL_SEQUENCE, RED, 0, false)
+    , FaultLabel(WATCHDOG_CAUSED_LAST_RESET, RED, 0, false)
+    , FaultLabel(CONFIG_READ_ERROR, RED, 0, false)
+    , FaultLabel(RAIL_UNDER_VOLTAGE_LOCK_OUT, RED, 0, false)
+    , FaultLabel(DESATURATION_FAULT, RED, 0, false)
+})
+, limitList_({FaultLabel(OUTPUT_VOLTAGE_PWM_LIMIT, YELLOW, 1, false)
+              , FaultLabel(MOTOR_CURRENT_LIMIT, YELLOW, 1, false)
+              , FaultLabel(VELOCITY_LIMIT, YELLOW, 1, false)
+              , FaultLabel(BUS_CURRENT_LIMIT, YELLOW, 1, false)
+              , FaultLabel(BUS_VOLTAGE_UPPER_LIMIT, YELLOW, 1, false)
+              , FaultLabel(BUS_VOLTAGE_LOWER_LIMIT, YELLOW, 1, false)
+              , FaultLabel(IPM_MOTOR_TEMPERATURE_LIMIT, YELLOW, 1, false)
+             })
+{
+}
+
+FaultList::~FaultList()
+{
+}
+
+FaultLabel FaultList::getHighestActivePriorityLabel() const
```

Comment:
This function should just return the highest active priority, and then another function `setHighestActivePriorityLabel` can have the logic for determining it.